hotpdf/encodings/decoder.py

```diff
@@ -0,0 +1,33 @@
+import re
+from typing import Optional
+
+from .types import EncodingTypes
+
+
+class Decoder:
+    """If there are no embedded fonts, pdfminer.six has issues with
+    mapping cid values to their specific unicode characters.
+    In that case, we manually override the (cid:int) values
+    """
+
+    __cid_mapping: dict[int, str] = {}
+    initialised: bool = False
+
+    def __init__(self, charset: Optional[EncodingTypes] = None) -> None:
+        if not charset:
+            return
+        if charset.value in EncodingTypes._value2member_map_:
+            self.initialised = True
+        if charset == EncodingTypes.LATIN:
+            from .mappings.latin import CID_TO_STR
+
+            self.__cid_mapping = CID_TO_STR
+
+    def cid_str_to_str(self, cid_str: str) -> str:
+        """Converts a (cid:int) notation to it's corresponding charset unicode
+        In case there's no mapping, return a blank string
+        """
+        cid_digit = re.search(r"\d+", cid_str)
+        if not cid_digit:
+            return cid_str
+        return self.__cid_mapping.get(int(cid_digit.group()), "")
```

Comment:
Possibly in the future, this fallback behavior `"` might be configured. Maybe we might not skip these symbols, because it's also space for errors.

Reply:
Yeah - but this code isn't reached if we don't pass the charset. 
What do you think about the `skip_unknown_chars=True` flag in the Decoder object? So we don't pollute the HotPdf initialisation.

`Decoder(charset: dict[int, str], skip_unknown_chars: bool)` ?

Reply:
I opened an issue for this. We can add it with the custom decoder idea (and make a minor release. For now let's have a non breaking change)

https://github.com/weareprestatech/hotpdf/issues/112